src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/SQLClause.java

```diff
@@ -47,8 +49,12 @@
      */
     abstract void rewrite(Scope scope);
 
-    /** Replace expr by nested(expr) and set pointer in parent properly */
     SQLMethodInvokeExpr replaceByNestedFunction(SQLExpr expr) {
+        return replaceByNestedFunction(expr, null);
```

Comment:
Because `nestedFunc` is returned, could we do this in the opposite way to avoid passing `null` explicitly?

```
    SQLMethodInvokeExpr replaceByNestedFunction(SQLExpr expr, String path) {
        SQLMethodInvokeExpr nestedFunc = replaceByNestedFunction(expr);
        nestedFunc.getParameters().add(1, new SQLCharExpr(path));
        return nestedFunc;
    }

    SQLMethodInvokeExpr replaceByNestedFunction(SQLExpr expr) {
       // same as original...
    }

Reply:
Yes, this seems much better.

Reply:
Done.